### PR TITLE
Firehose-cosmos documentation

### PR DESCRIPTION
**Pull request by lavabyrd**
https://figmentio.atlassian.net/browse/GRAPH-255

Seeing this live is not easy. If you wish you can install Hugo using the following guide, then checkout this branch and use `hugo serve`

```
git clone https://github.com/gohugoio/hugo.git
cd hugo
go install -v --tags extended
```

### CODE REVIEW

File: content/operate/guides/cosmos/syncing.md [hunk 1]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+## Install firehose-cosmos
+
+`firehose-cosmos`, is an application that runs a few small, isolated processes,
+that together form the `Firehose` stack. A thorough discussion of the [Concepts & Architecture]({{< ref "/operate/concepts" >}})
+is discussed elsewhere. Needless to say, you must run `firehose-cosmos` to run a `Firehose` locally.
+
+You can download the latest version of `firehose-cosmos` [here](https://github.com/figment-networks/firehose-cosmos/releases),
+however we recommend you use the dockerfile located in the `firehose-cosmos` repository located [here](https://github.com/figment-networks/firehose-cosmos)
+
+Once downloaded you must unzip the bundle or alternatively you can clone the repository and run `make install` to install it locally.
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/syncing.md]: I would make a code block here with command. Not sure what is meant by unzip the bundle. The release is not a zip unless someone decide to download the source but I would guess people wanting to build from source would clone the repo instead. We should probably publish a built docker image and use that in doc.

File: content/operate/guides/cosmos/syncing.md [hunk 2]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+    ingestor-node-path: path/to/gaiad_or_osmosisd
+    ingestor-node-args: start --x-crisis-skip-assert-invariants
+    ingestor-node-env: "KEY=VALUE,KEY=VALUE"
+```
+
+---
+
+### Logs input mode
+
+It's possible to run the firehose ingestor from the static logs, mostly for development/testing purposes.
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/syncing.md]: We should add also an example for the config needed to gaiad as above

File: content/operate/guides/cosmos/syncing.md [hunk 3]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+    ingestor-mode: logs
+    ingestor-logs-dir: /path/to/logs/dir
+
+    # Configure the pattern if not using .log extension
+    # ingestor-logs-pattern: *.log
+```
+
+---
+
+## Binary configuration
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/syncing.md]: Again ? Already talked about above.

File: content/operate/guides/cosmos/syncing.md [hunk 4]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+###       Extractor Configuration Options           ###
+#######################################################
+[extractor]
+enabled = true
+output_file = "stdout"
+```
+
+## Syncing your chain
+
+One you begin running `firehose-cosmos`, you will begin to see your blocks streaming through if it is successfully instrumented.
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/syncing.md:82]: Add code block for start command ? Should we also talk about firehose.yml or it's already covered elsewhere in the doc? If so reference to it.

File: content/operate/guides/cosmos/syncing.md [hunk 5]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+2022-05-30T22:39:39.414+0100 (dfuse) All apps terminated gracefully (launcher/launcher.go:273)
+```
+
+---
+
+## Overview and Explanation
+
+The `ingestor` process is responsible for reading events from the node process either through Flat files, or STDOUT depending on configuration in the Extractor process
+
+The `merger` process will either write individual block data into separate files called one-block files,
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/syncing.md]: Sure about that? Isn't the ingestor writting the one block file?

File: content/operate/guides/cosmos/system-requirements.md [hunk 6]
```diff
@@ -4,4 +4,30 @@ title: System Requirements
...
+* **The length of history**: whether or not you are serving all the blocks through the firehose
+
+The CPU/RAM requirements will depend on these factors:
+
+* **High Availability**: highly available deployments will require **2 times the resources** (or more) listed in the following examples, as a general rule.
+* **Throughput of queries**: the Firehose stack is built for horizontal scalability, the more requests per second you want to fulfill, the larger the deployment, the more CPU/RAM you will need to allocate to your cluster.
+
+{{< alert type="note" >}}
+These stats are from March 2022, but chains of similar density and similar age will behave similarly.
+{{< /alert >}}
```
Comment (Seb-figment) [inline content/operate/guides/cosmos/system-requirements.md]: I would keep the note saying this section is incomplete as for Etherium

File: content/operate/guides/cosmos/syncing.md [hunk 7]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+    - firehose
+  flags:
+    # Common flags
+    common-first-streamable-block: 1
+
+    # Ingestor specific flags
+    ingestor-mode: node
+    ingestor-node-path: path/to/gaiad_or_osmosisd
+    ingestor-node-args: start --x-crisis-skip-assert-invariants
+    ingestor-node-env: "KEY=VALUE,KEY=VALUE"
```
Comment (sosedoff) [inline content/operate/guides/cosmos/syncing.md]: We should get rid of this example line `ingestor-node-env`

File: content/operate/guides/cosmos/syncing.md [hunk 8]
```diff
@@ -4,6 +4,281 @@ title: Syncing
...
+If this is the first time you have run the node, this will start the node from genesis, so give it some time until it start syncing.
+You may check on the node's status (if its running locally) by opening `http://localhost:26657/status` in your browser.
+
+To test if firehose is ready to stream blocks, you can use the grpcurl command:
+
+```bash
+grpcurl -plaintext localhost:9030 sf.firehose.v1.Stream.Blocks | jq
+```
+
+Make sure you have both `grpcurl` and `jq` installed. If you don't, you should be able to find them on your preferred package manager
```
Comment (sosedoff) [inline content/operate/guides/cosmos/syncing.md]: Add links to the github repos for these tools as now all of them are available via package managers.

File: content/operate/guides/cosmos/syncing.md (hunk 5 above)
Comment (sosedoff) [reply content/operate/guides/cosmos/syncing.md]: Ditto

File: content/operate/guides/cosmos/syncing.md [hunk 9]
```diff
@@ -4,6 +4,242 @@ title: Syncing
 showH2InSideNav: true
 ---
 
-In this guide, we'll show you how to use [Firehose](/operate/concepts/) to sync and stream Cosmos chains.
+Below, we'll show you how to use [Firehose](/operate/concepts/) to sync and stream Cosmos Based Chains.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: I would keep "Cosmos chains" here.

File: content/operate/guides/cosmos/syncing.md [hunk 10]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+## Install firehose-cosmos
+
+`firehose-cosmos`, is an application that runs a few small, isolated processes,
+that together form the `Firehose` stack. A thorough discussion of the [Concepts & Architecture]({{< ref "/operate/concepts" >}})
+is discussed elsewhere. Needless to say, you must run `firehose-cosmos` to run a `Firehose` locally.
+
+You can download the latest version of `firehose-cosmos` [here](https://github.com/figment-networks/firehose-cosmos/releases),
+however we recommend you use the Dockerfile provided in the `firehose-cosmos` repository located [here](https://github.com/figment-networks/firehose-cosmos).
+
+As of the time of writing this, the latest release is v0.4.0, you can get a copy of this from [Figment's Docker Hub](https://hub.docker.com/r/figmentnetworks/firehose-cosmos/tags). The latest release will always be listed on the [Github Releases Page](https://github.com/figment-networks/firehose-cosmos/releases). Ensure you always use the most up to version of `firehose-cosmos` to ensure you have the latest functionality.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: > most up to version

Should it be "most up to date version"?

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: > most up to version

Should it be "most up to date version"?

File: content/operate/guides/cosmos/syncing.md [hunk 11]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+```bash
+firehose-cosmos --version
+# or
+docker run --rm -it figmentnetworks/firehose-cosmos:0.4.0 /app/firehose --version
+```
+---
+
+## Install and run instrumented nodes
+
+In order to index Cosmos based nodes with `firehose-cosmos`, you will need to insure you are using the correctly modified binaries for the chain you are targetting. Figment provide these for CosmosHub and you can find them on the `Gaia-DM` releases page located [here](https://github.com/figment-networks/gaia-dm/releases). Alternatively, they also provide the prebuilt Docker Images which you can find listed on [their Dockerhub](https://hub.docker.com/r/figmentnetworks/firehose-cosmos/tags). These prebuilt Dockerfiles also start the `firehose-cosmos` process for you. Below is an example for v7.0.4 of the Osmosis testnet. You can modify this to use whichever version of the chain node you are indexing.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: > Cosmos based nodes

Cosmos nodes?

> you can find them on the `Gaia-DM` releases

I would not mention the `Gaia-DM` name here, we might change it in the future.

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: > Cosmos based nodes

Cosmos nodes?

> you can find them on the `Gaia-DM` releases

I would remove the `Gaia-DM` name, we might change that in the future.

File: content/operate/guides/cosmos/syncing.md [hunk 12]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+```ini
+#######################################################
+###       Extractor Configuration Options           ###
+#######################################################
+[extractor]
+enabled = true
+output_file = "stdout"
+```
+
+## Firehose-Cosmos Configuration
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: I would use either "Firehose Cosmos" or firehose-cosmos.

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: I would use either "Firehose Cosmos" or firehose-cosmos.

File: content/operate/guides/cosmos/syncing.md [hunk 13]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+
+If you wish to use a configuration file instead of setting all CLI flags, you can pass it with a `-c ./configfile.ini` argument.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: Is `configfile.ini` the right name? I thought it was a YAML file.

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: Is `configfile.ini` the right name? I thought it was a YAML file.

File: content/operate/guides/cosmos/syncing.md [hunk 14]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+If this is the first time you have run the node, this will start the node from genesis, so give it some time until it start syncing.
+You may check on the node's status (if its running locally) by opening `http://localhost:26657/status` in your browser.
+
+To test if firehose is ready to stream blocks, you can use the grpcurl command:
+
+```bash
+grpcurl -plaintext localhost:9030 sf.firehose.v1.Stream.Blocks | jq
+```
+
+Make sure you have both [grpcurl](https://github.com/fullstorydev/grpcurl) and [jq](https://github.com/stedolan/jq) installed. If you don't, you should be able to find them on your preferred package manager
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md:92]: Missing full stop.

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: Missing full stop.

File: content/operate/guides/cosmos/syncing.md [hunk 15]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+
+## Overview and Explanation
+
+The `ingestor` process consumes data from the node and saves it as one-blocks files.
+
+The `merger` process downloads these single block files and creates bigger 100 blocks files inside the `merged-block` folder with a naming convention that matches the blocks inside the file. e.g. `0005200700.dbin.zst` contains the blocks from `5200700 - 5200799`.
+
+The `relayer` process uses gRPC to connect to the `ingestor` process and consume real-time block updates. It can also use the `merger` as a back-up mechanism in the case there are missing files.
+
+The `firehose` process will connect to the `relayer(s)` via gRPC and to the data store in order to provide a live feed of blocks in a joined manner. The end consumer can connect to it via gRPC to read from the stream. It uses the merged-block files created by the `merger` process to serve this data to the services downstream.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: This part is the same for all Firehoses, we could link to a page where it is explained (or skip it entirely).

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: This part is the same for all Firehoses, we could link to a page where it is explained (or skip it entirely).

File: content/operate/guides/cosmos/syncing.md [hunk 16]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+
+---
+
+## What's next
+
+Congratulations! You're now streaming blocks data from a Cosmos based chain.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: Cosmos chain?

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md]: Cosmos chain?

File: content/operate/guides/cosmos/syncing.md [hunk 17]
```diff
@@ -4,6 +4,242 @@ title: Syncing
...
+
+At this point, Cosmos networks, such as CosmosHub and Osmosis are yours to discover. Slice and dice this data to your heart's content.
```
Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md:239]: I like this part :)

Comment (pienkowb) [inline content/operate/guides/cosmos/syncing.md:239]: I like this part :)

File: content/operate/guides/cosmos/system-requirements.md (hunk 6 above)
Comment (pienkowb) [reply content/operate/guides/cosmos/system-requirements.md]: I agree. We could also remove this page (if it's just placeholder and we don't provide any information here).

File: content/operate/guides/cosmos/syncing.md (hunk 12 above)
Comment (lavabyrd) [reply content/operate/guides/cosmos/syncing.md]: duplicate comment

File: content/operate/guides/cosmos/syncing.md (hunk 13 above)
Comment (lavabyrd) [reply content/operate/guides/cosmos/syncing.md]: I believe you can use either 🤔 i'll update to YAML to match the readme though

**Review by pienkowb — APPROVED** [review]
LGTM 👍

**Comment by lavabyrd** [comment]
Will close this PR and merge directly from the cosmos branch the SF's master

**Pull request closed without merging**